File: backend/src/main/com/rot/access/dtos/AccessDto.kt
package com.rot.access.dtos

import java.time.OffsetDateTime

class AccessDto {
    var accessToken: String? = null
    var accessTokenExpiresAt: OffsetDateTime? = null
    var refreshToken: String? = null
    var refreshTokenExpiresAt: OffsetDateTime? = null
}

File: frontend/src/common/models/access/AccessDto.ts
import { Transform } from 'class-transformer';
import { Dayjs } from 'dayjs';
import { transformerDateTime } from 'src/common/models/models';

export class AccessDto {
  accessToken!: string;
  refreshToken!: string;

  @Transform(transformerDateTime)
  accessTokenExpiresAt!: Dayjs;

  @Transform(transformerDateTime)
  refreshTokenExpiresAt!: Dayjs;
}
